src/main/webapp/WEB-INF/tags/instructor/courseStudentDetails/studentProfile.tag

```diff
@@ -23,7 +23,18 @@
             <tr>
               <td class="text-bold">Short Name (Gender)</td>
               <td>${empty student.shortName ? none : fn:escapeXml(student.shortName)}
-                (<i> ${student.gender == other ? noneForGender : fn:escapeXml(student.gender)} </i>)
```

Comment:
Why not just replace `fn:escapeXml(...)` to `student.displayedGenderType`? I believe it's the same effect with a much shorter change.

Reply:
@wkurniawan07 Sir Simply replacing the fn:escapeXml(. . . ) to student,displaydedGenderType will not give the desired solution as you can see I am using text-muted format when Gender is Not Specified that is in the case of null . but here it is in the case when gender is "other" . I have slightly modified to fit it according to my logic 
${student.displayedGenderType == Not specified ? noneForGender : student.displayedGenderType . but still It is not giving the desired result It Only give desired result when I am using my logic of using JSTl condition of <c;choose> and all . So it do not seems beneficial to use the appropach suggested by you

Reply:
@wkurniawan07  Sir so this approach of using <:choose> <:/choose>  is not going to do any problem.

Reply:
> student.displayedGenderType == Not specified

What are you trying to do here? Space in variable name is a bad idea, if even allowed at all.

Reply:
@wkurniawan07 sir I am using from this Const.STUDENT_PROFILE_FIELD_NOT_FILLED = "Not Specified"

Reply:
first I have set the variable as 

<c:set var="noneForGender"><span class="text-muted"><%= Const.STUDENT_PROFILE_FIELD_NOT_FILLED %></span></c:set>

 <c:set var="Not Specified"><%= Const.STUDENT_PROFILE_FIELD_NOT_FILLED %></c:set>

and then used
(<i>  ${student.displayedGenderType} == Not Specified ? noneForGender : ${student.displayedGenderType}</i>)
but it is not working .
if I use variable without space then also problem will persist  .


but when I used the code 

                <c:choose>
                    <c:when test="${student.displayedGenderType} == Const.STUDENT_PROFILE_FIELD_NOT_FILLED">
                      <span class="text-muted">
                         ${student.displayedGenderType}
                      </span>
                     </c:when>
                      <c:otherwise>
                          ${student.displayedGenderType}
                      </c:otherwise>
                   </c:choose>

problem is solved

Reply:
@wkurniawan07  sir Giving the whole scenerio in nutshell I think merging the current approach as I am using will not do any harm . as Idea suggested by you will only change way of doing the code not the logic .

Reply:
> `${student.displayedGenderType} == Not Specified ? noneForGender : ${student.displayedGenderType}`

I will assume you mean `${student.displayedGenderType == Not Specified ? noneForGender : student.displayedGenderType}`. The mistake here is obvious: `Not Specified`. You're either trying `student.displayedGenderType == "Not Specified"` or `student.displayedGenderType == notSpecified`. Actually, why not just use the previous condition `student.gender == other`? From the very beginning I only suggested to change the `fn:escapeXml(...)`.

> I think merging the current approach as I am using will not do any harm

The problem is you're writing more code than necessary, and any code in the code base is a maintenance burden. Let me give you this analogy: if you can get 2 by simply 1 + 1, will you accept a solution such as 1 + 5 + 4 - 3 - 5?

Reply:
@wkurniawan07 sir I have made appropriate changes here alsoo as you can see

Reply:
@wkurniawan07  I can't use the previous condition of student.gender == other due to
1) As I am using the logic when the Gender field is null then Gender is Not Specified and it should be displayed in text-muted form else it should be displayed in normal form 
2) As In whole program Gender are MALE/FEMALE or OTHER (That is in uppercase)so I have used the Method getdisplayedGenderType() to convert this uppercase Gender in lower case so that it can be displayed to the user without noticeable change .It would not look good if we Display the gender of the person as "MALE" so instead I changed it to lowercase so that it can look good to user so case of directly calling getgender() method by using student.gender rules out 
3) I have used the logic if not male/female then other as suggested by you when I started working on this issue